tensorflow/stream_executor/cuda/cuda_dnn.cc

```diff
@@ -1225,9 +1242,40 @@ class CudnnRnnSequenceTensorDescriptor
         /*strideA=*/strides));
     return CudnnRnnSequenceTensorDescriptor(parent, seq_length, batch_size,
                                             data_size, data_type,
-                                            std::move(tensor_desc));
+                                            tensor_desc);
   }
 
+  static port::StatusOr<CudnnRnnSequenceTensorDescriptor> CreateVar(
+          CUDAExecutor* parent, int seq_length, int batch_size, int data_size, const int32_t* sequence_lengths,
+          cudnnDataType_t data_type) {
+        CHECK_GT(seq_length, 0);
+        TensorDescriptor tensor_desc = CreateTensorDescriptor();
+        {
+            int dims[] = {batch_size, data_size, 1};
+            int strides[] = {data_size, 1, 1};
+            RETURN_IF_CUDNN_ERROR(cudnnSetTensorNdDescriptor(
+                /*tensorDesc=*/tensor_desc.get(), /*dataType=*/data_type,
+                /*nbDims=*/3, /*dimA=*/dims,
+                /*strideA=*/strides));
+        }
+        auto descriptors = std::vector<TensorDescriptor>(0);
+        auto handles = std::vector<cudnnTensorDescriptor_t>(seq_length);
+        int pos = batch_size-1;
+        for(int i=0; i< seq_length; i++){
+            while(sequence_lengths[pos] <= i && pos > 0){
+                pos--;
+            }
+            auto td = CreateTensorDescriptor();
```

Comment:
perhaps ```const``` if td has a well written move constructor (or default copy and move can be used just fine), I saw that because i see you std::move it after.